src/daft-local-execution/src/sinks/write.rs

```diff
@@ -83,16 +120,26 @@ impl BlockingSink for WriteSink {
         mut state: Box<dyn BlockingSinkState>,
         spawner: &ExecutionTaskSpawner,
     ) -> BlockingSinkSinkResult {
+        let builder = spawner.runtime_context.builder.clone();
+
         spawner
             .spawn(
                 async move {
-                    state
+                    let bytes_written = state
                         .as_any_mut()
                         .downcast_mut::<WriteState>()
                         .expect("WriteSink should have WriteState")
                         .writer
                         .write(input)
                         .await?;
+
+                    builder
```

Comment:
Example of a custom stats tracker and how to update it within the operator. Wan't to keep it simple without too much magic. Lmk if you have different opinions